src/components/Timer/timer.js

```diff
@@ -0,0 +1,99 @@
+import { Component } from 'react';
+import '../../style/timer.css';
+
+export default class Timer extends Component {
+  constructor(props) {
+    super(props);
+
+    // Initial state with Pomodoro mode, initial Pomodoro duration, and counter
+    this.state = {
+      mode: 'Pomodoro',
+      time: props.timer.pomodoro * 60, // Time in seconds for Pomodoro
+      isRunning: false,
+      pomodoroCount: 0
+    };
+
+    // Interval variable for updating the timer
+    this.timerInterval = null;
+  }
+
+  stopTimer = () => {
+    clearInterval(this.timerInterval);
+    this.setState({ isRunning: false });
+  };
+
+  startTimer = () => {
+    if (this.state.isRunning) return;
+
+    this.setState({ isRunning: true });
+
+    this.timerInterval = setInterval(() => {
+      this.setState((prevState) => ({
+        time: prevState.time - 1,
+      }));
+
+      if (this.state.time <= 0) {
+        this.stopTimer();
+        this.setState({ time: 0 });
+
+        if (this.state.mode === 'Pomodoro') {
+          // Increment the Pomodoro count after completing a Pomodoro session
+          this.setState((prevState) => ({
+            mode: 'Break',
+            time: this.props.timer.break * 60,
+            pomodoroCount: prevState.pomodoroCount + 1,
+          }));
+          this.startTimer();
+        } else {
+          // Reset to Pomodoro mode after the break
+          this.setState({
+            mode: 'Pomodoro',
+            time: this.props.timer.pomodoro * 60,
+          });
+        }
+      }
+    }, 1000);
+  };
+
+  restartTimer = () => {
+    this.stopTimer();
+    this.setState({
+      mode: 'Pomodoro',
+      time: this.props.timer.pomodoro * 60,
+      isRunning: false,
+    });
+  };
+
+
+  render() {
+    const { mode, time, pomodoroCount } = this.state;
+
+    return (
+      <div className="timer-container">
+        <div className="timer-mode">{mode}</div>
+        <div className="timer-time">
+          {Math.floor(time / 60)}:{(time % 60).toLocaleString('en-US', { minimumIntegerDigits: 2 })}
+        </div>
+
+        <span className="badge text-bg-success pomodoro-counter">
+          Completed pomodoros: {pomodoroCount}
+        </span>
+
+        <div className="timer-buttons">
+          {this.state.isRunning ? (
+            <button onClick={this.stopTimer} type="button"className="btn btn-light">
+              <span className="material-symbols-outlined">pause</span>
+            </button>
+          ) : (
```

Comment:
This is the right solution, but you made the changes in a really outdated branch. This is not acceptable for main branch. You should merge the main into this branch before it's acceptable.

Reply:
This is now solved.